src/transactions/feed/JumpstartTransactionDetailsScreen.tsx

```diff
@@ -0,0 +1,320 @@
+  })
+
+  if (parsedLogs.length != 1) {
+    throw new Error('Unexpected number of matching logs')
+  }
+
+  const { beneficiary, index } = parsedLogs[0].args
+
+  Logger.debug('Decoded event', { beneficiary, index })
+
+  const erc20Claim = await viemClient.readContract({
+    address: jumpstartContractAddress as Address,
+    abi: walletJumpstart.abi,
+    functionName: 'erc20Claims',
+    args: [beneficiary, index],
+  })
+
+  const claimed = erc20Claim[3]
+
+  Logger.debug(`Reward deposited in ${transactionHash} ${claimed ? 'was' : 'was NOT'} claimed`)
+
+  return { beneficiary, index: Number(index), claimed }
+}
+
+function JumpstartTransactionDetailsScreen({ route }: Props) {
+  const { transaction } = route.params
+  const { t } = useTranslation()
+
+  const [reclaimTx, setReclaimTx] = useState<SerializableTransactionRequest | null>(null)
+  const transactionString = useMemo(() => (reclaimTx ? JSON.stringify(reclaimTx) : ''), [reclaimTx])
+
+  const transactionTokenInfo = useTokenInfo(transaction.amount.tokenId)
+  const parsedAmount = new BigNumber(transaction.amount.value).abs()
+  const token = useTokenInfo(transaction.amount.tokenId)
+  const isDeposit = transaction.type === TokenTransactionTypeV2.Sent
+  const jumpstartContractAddress = transaction.address
+
+  const walletAddress = useSelector(walletAddressSelector)
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const title =
+    transaction.type === TokenTransactionTypeV2.Sent
+      ? t('feedItemJumpstartTitle')
+      : t('feedItemJumpstartReceivedSubtitle')
+
+  const fetchClaimData = useAsync(
+    async () => {
+      if (!isDeposit) {
+        return
+      }
+      return await getClaimDataAndStatus(
+        jumpstartContractAddress as Address,
+        transaction.networkId,
+        transaction.transactionHash as Hash
+      )
+    },
+    [],
+    {
+      onError: (error) => {
+        // TODO: Show error message in the UI
+        Logger.error(TAG, 'Failed to fetch escrow data', error)
+      },
+    }
+  )
+
+  if (!token) {
+    // should never happen
+    Logger.error(TAG, 'Token is undefined')
+    return null
+  }
+
+  const onReclaimPress = () => {
+    if (!fetchClaimData.result) {
+      Logger.error(TAG, 'No escrow data found when trying to reclaim')
+      return
+    }
+
+    const { beneficiary, index } = fetchClaimData.result
+
+    setReclaimTx({
+      from: walletAddress as Address,
+      to: jumpstartContractAddress as Address,
+      value: '0',
+      data: encodeFunctionData({
+        abi: walletJumpstart.abi,
+        functionName: 'reclaimERC20',
+        args: [beneficiary, BigInt(index)],
+      }),
+      // TODO: Estimate
+      gas: '130000',
+      maxFeePerGas: '1000000000',
+    })
+
+    bottomSheetRef.current?.snapToIndex(0)
+  }
+
+  const isClaimed = fetchClaimData.result?.claimed
+
+  const reclaimButton = (
+    <View style={styles.buttonContainer}>
+      <Button
+        showLoading={!fetchClaimData.result}
```

Comment:
curious why this is not loading when `fetchClaimData.loading` only?

Reply:
Yes, I went back and forth on this, when there is an error fetching the data should we show a disable "Reclaim" or show a disable spinner?
I changed it to `loading` in the next PR, but I'll bring that change here.

I will show a disable "Reclaim" and also the error message, and then if the user dismisses the error message, I'll trigger a new fetch. What do you think?
It's that or just leave the "Reclaim" as disable (so the user has to go back and enter again to try to reclaim again).

What I like of leaving the spinner is that if there is an error when fetching the data, we don't know whether the escrow was claimed or not, so showing "Reclaim" as disable gives a hint that it wasn't claim yet (but I think we can live with this). Or should I ask Kayla for some feedback for this case, maybe we can to change the wording of the button in that case? 🙈

Reply:
aw man, tricky. i think we should show loaders only when there is a background load happening, when loaders are shown for too long then it looks like the app is buggy. so i think for the error state, we shouldn't show the loader.

so the question is what should we show 😅 i think it's probably find to leave it as a disabled "Reclaim" state. let's track how many times users are hitting this error to see if we want to spend more time on this - fwiw i think the error popup is already enough UI feedback that something has gone wrong, so the disabled button label shouldn't be too consequential.

i also think it's okay to not retry the fetch, and let users come back to the screen. if i understand the fetch function, errors are likely _not_ to be transient so adding a retry doesn't really increase the chances of users being able to proceed? so seems not worth the effort for now